adapter/internal/proxyserver/handler.go

```diff
@@ -0,0 +1,217 @@
+func (s *Server) HTTPPostToExternalMSHandler(msUniqueIdentifier string) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		buildRequestForExtMS(s.client, w, r, http.MethodPost, r.Body, msUniqueIdentifier)
+	})
+}
+
+// HTTPPatchToExternalMSHandler function handles post requests sending to external microservices
+func (s *Server) HTTPPatchToExternalMSHandler(msUniqueIdentifier string) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		buildRequestForExtMS(s.client, w, r, http.MethodPatch, r.Body, msUniqueIdentifier)
+	})
+}
+
+func buildRequestToGA(client *http.Client, w http.ResponseWriter, r *http.Request, httpMethod string, body io.Reader) {
+	conf, error := config.ReadConfigs()
+
+	if error != nil {
+		logger.LoggerMgw.Error("Error occured while reading config.")
+		return
+	}
+	var requestURL = conf.GlobalAdapter.ServiceURL + r.RequestURI
+
+	sendHTTPRequestToGA(client, conf, requestURL, w, httpMethod, body)
+}
+
+func buildRequestForExtMS(client *http.Client, w http.ResponseWriter, r *http.Request, httpMethod string, body io.Reader, msUniqueIdentifier string) {
+	conf, error := config.ReadConfigs()
+
+	if error != nil {
+		logger.LoggerMgw.Error("Error occured while reading config.")
+		return
+	}
+	var msServiceURL string
+
+	// Handle a new microservice by adding corresponding case.
+	switch msUniqueIdentifier {
+	case moeisfUniqueIdentifier:
+		msServiceURL = conf.MoesifMicroservice.ServiceURL
+	default:
+		logger.LoggerMgw.Error("Unrecognizable microservice identifier has been passed: ", msUniqueIdentifier)
+		return
+	}
+	var requestURL = msServiceURL + r.RequestURI
+
+	sendHTTPRequestToExtMS(client, conf, requestURL, w, httpMethod, body, msUniqueIdentifier)
+}
+
+func sendHTTPRequestToGA(client *http.Client, conf *config.Config, requestURL string, w http.ResponseWriter, httpMethod string, body io.Reader) {
+	logger.LoggerMgw.Debug("Request URL for Global Adapter: ", requestURL)
+	req, err := http.NewRequest(httpMethod, requestURL, body)
+
+	if err != nil {
+		w.WriteHeader(500)
+		logger.LoggerMgw.Error("Error occurred while constructing http request to Global Adapter ", err)
+		return
+	}
+	req.SetBasicAuth(conf.GlobalAdapter.Username, conf.GlobalAdapter.Password)
+
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+
+	if err != nil {
+		w.WriteHeader(500)
+		logger.LoggerMgw.Error("Error occurred while connecting to Global Adapter ", err)
+		return
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		bodyBytes, _ := ioutil.ReadAll(resp.Body)
+		bodyString := string(bodyBytes)
+		w.WriteHeader(resp.StatusCode)
+		fmt.Fprintf(w, "%s", bodyString)
+	}
+
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+	bodyString := string(bodyBytes)
+	fmt.Fprintf(w, "%s", bodyString)
```

Comment:
This is not relevant to the scope of this PR, I think we should need some sort of a retry mechanism and probably it would be better to introduce that at the enforcer level. Just for your information.

Reply:
Ack.